src/grpc_server.cc

```diff
@@ -2309,23 +2305,34 @@ class InferHandlerState {
     complete_ = false;
     request_.Clear();
     response_queue_->Reset();
+#ifdef TRITON_ENABLE_TRACING
+    // Clear trace_timestamps_ here so they do not grow indefinitely since
+    // states are re-used for performance.
+    ClearTraceTimestamps();
+#endif  // TRITON_ENABLE_TRACING
   }
 
   void Release()
   {
     context_ = nullptr;
+#ifdef TRITON_ENABLE_TRACING
+    ClearTraceTimestamps();
+#endif  // TRITON_ENABLE_TRACING
```

Comment:
You can drop the `ifdef` here and above as you have done so in `ClearTraceTimestamps()`, this invocation should be optimized out if tracing is disabled.